### PR TITLE
csclient: add the possibility to disable stats.

### PR DESCRIPTION
**Pull request by frankban**
So that it is possible to retrieve archives for tests.

### CODE REVIEW

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/821/
Test PASSed.

File: csclient/csclient_test.go [hunk 1]
```diff
@@ -185,6 +186,21 @@ func (s *suite) TestGet(c *gc.C) {
...
+	stats.CheckCounterSum(c, s.store, key, false, 1)
+}
+
+func (s *suite) TestGetArchiveWithStatsDisabled(c *gc.C) {
+	s.client.DisableStats()
+	key := s.checkGetArchive(c)
+
+	// Check that the downloads count for the entity has not been updated.
+	stats.CheckCounterSum(c, s.store, key, false, 0)
+}
```
Comment (bac) [inline csclient/csclient_test.go:201]: I think both of these tests would be more demonstrative if they started with a non-zero value.

**Comment by bac** [comment]
:+1: with one suggestion

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/822/
Test PASSed.

File: csclient/csclient_test.go (hunk 1 above)
Comment (frankban) [inline csclient/csclient_test.go:201]: This is a server side functionality, and those tests are just here to demonstrate DisableStats, so an initial 0 value is good enough I guess.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/823/
Test PASSed.

**Comment by frankban** [comment]
Thanks for the reviews Brad and Martin!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**